components/TriviaQuestions/index.js

```diff
@@ -2,84 +2,98 @@ import React, { useState, useContext } from 'react';
 import { View, StyleSheet, TouchableOpacity } from 'react-native';
 import base64 from 'react-native-base64';
 import { Text } from 'react-native-elements';
+import EvilIcon from 'react-native-vector-icons/EvilIcons';
 
-import SuccessFail from '../SuccessFail';
 import { AppContext } from 'context';
 import Modal from '../Modal';
 import colors from 'consts/colors';
 
 const TriviaQuestions = ({ data }) => {
-  //@TODO Need to refactor this logic so this only scrambles once
-  function shuffleArray(array) {
-    let i = array.length - 1;
-    for (; i > 0; i--) {
-      const j = Math.floor(Math.random() * (i + 1));
-      const temp = array[i];
-      array[i] = array[j];
-      array[j] = temp;
-    }
-    return array;
-  }
   const { getTriviaData, id, difficulty } = useContext(AppContext);
-  console.log(id, difficulty);
 
   const [counter, setCounter] = useState(1);
   const [selected, setSelected] = useState('');
-  const [score, setScore] = useState(0);
-  console.log(score);
+  const [clicked, setClicked] = useState(false);
+  const [correct, setCorrect] = useState();
 
-  const handleSubmit = () => {
+  const handleClick = (d, i) => {
+    setCorrect();
+    setSelected(d);
+    setClicked(true);
+  };
+
+  const nextQuestion = () => {
+    setClicked(false);
     setSelected('');
-    setCounter(counter + 1);
     getTriviaData(id, difficulty);
+    setCounter(counter + 1);
+  };
+  const handleCorrectResult = () => {
+    return (
+      <View>
+        <EvilIcon size="20" name="check" color={colors.primaryColor} />
```

Comment:
My app was crashing after answering a question saying this value needs to be a number instead of a string - doing so did indeed fix it for me

Reply:
all set